R-package/src/install.libs.R

```diff
@@ -52,22 +57,28 @@ if (!use_precompile) {
       build_cmd <- "mingw32-make.exe _lightgbm"
       system(paste0(cmake_cmd, " ..")) # Must build twice for Windows due sh.exe in Rtools
     } else {
-      try_vs <- 0L
```

Comment:
Why this initialization was removed?

Reply:
This isn't necessary, I think maybe it is a misunderstanding about how `system()` works. You don't need to allocate an int and then fill it. `system()` will just return it.

Reply:
OK, so in R variable defined in nested scope remains visible in outer scope later, right?

Reply:
That's right. Python is the same way

```python
for i in range(5):
    x = i
```

If you think that isn't obvious though I can restore this! But I'd change it to `1L` so that that default assumption is "VS didn't work" instead of "assume it worked"

Reply:
I see, thanks for the explanation!

I'd better leave it up to other reviewers.

Reply:
no prob! @Laurae2 or @guolinke what do you think?

Reply:
gently ping @guolinke

Reply:
I think it could be remvoed.

Reply:
great, thank you